### PR TITLE
[test] Update Cell.swift to use new mangling

### PR DESCRIPTION
**Pull request by Azoy**
This PR: https://github.com/apple/swift/pull/72592 and this PR: https://github.com/apple/swift/pull/72692 raced with each other, so update this test to use the new mangling.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please smoke test

File: test/SILOptimizer/stdlib/Cell.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Builtin
 @frozen
 @_rawLayout(like: T)
 public struct Cell<T: ~Copyable>: ~Copyable {
-  // CHECK-LABEL: sil {{.*}} @$s4CellAAVAARiczrlE7addressSpyxGvg : $@convention(method) <T where T : ~Copyable> (@in_guaranteed Cell<T>) -> UnsafeMutablePointer<T> {
+  // CHECK-LABEL: sil {{.*}} @$s4CellAAVAARi_zrlE7addressSpyxGvg : $@convention(method) <T where T : ~Copyable> (@in_guaranteed Cell<T>) -> UnsafeMutablePointer<T> {
```
Comment (drodriguez) [inline test/SILOptimizer/stdlib/Cell.swift:8]: You also need to change the `end sil function` bits at the end of the blocks.

**Review by drodriguez — CHANGES_REQUESTED** [review]

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Review by drodriguez — APPROVED** [review]

**Comment by rintaro** [comment]
Force merging  to unblock bots without waiting for PR tests to finish. Confirmed the test case passed.

**Pull request merged**